src/components/TextField/TextField.tsx

```diff
@@ -315,15 +315,15 @@ export function TextField({
     clearTimeout(buttonPressTimer.current);
   }, []);
 
-  const handleButtonPress = useCallback(
-    (onChange: Function) => {
+  const handleButtonPress: SpinnerProps['onMouseDown'] = useCallback(
+    (onChange) => {
       const minInterval = 50;
       const decrementBy = 10;
       let interval = 200;
 
       const onChangeInterval = () => {
         if (interval > minInterval) interval -= decrementBy;
-        onChange();
+        onChange(0);
```

Comment:
Fixing up the types in Spinner highlighted that this was wrong - onChange should take a number as an argument. This is still not doing what we'd hope the code should do - holding the step buttons doesn't increment the value - but at least it is properly typed now. I briefly tried fixing this but didn't get it working. I think a proper fix would be better as a separate PR.